### PR TITLE
feat: add object detection with YOLO

### PR DESCRIPTION
**Pull request by CarlosCRG19**
## Changes

- Created `ObjectDetectionModel` abstraction and implementation using YOLO
- Expose `/detect-objects` endpoint to detect objects in an image

### Sample responde of POST /detect-objects endpoint
```
{
    "objects": [
        {
            "type": "button",
            "confidence": 0.8809658288955688,
            "bbox": {
                "x1": 1344.7420654296875,
                "y1": 106.3250503540039,
                "x2": 1429.515625,
                "y2": 146.4981231689453,
                "normalized_mid_point": [
                    0.8669555187225342,
                    0.14414092898368835
                ]
            }
        },
        {
            "type": "button",
            "confidence": 0.8692622780799866,
            "bbox": {
                "x1": 66.44298553466797,
                "y1": 160.290283203125,
                "x2": 182.43312072753906,
                "y2": 206.4077911376953,
                "normalized_mid_point": [
                    0.07777377963066101,
                    0.20906388759613037
                ]
            }
        },
        {
            "type": "text",
            "confidence": 0.8492657542228699,
            "bbox": {
                "x1": 924.9713745117188,
                "y1": 463.5417785644531,
                "x2": 1061.4976806640625,
                "y2": 477.2769775390625,
                "normalized_mid_point": [
                    0.620771586894989,
                    0.5363847017288208
                ]
            }
        },
    ]
}
```

### CODE REVIEW

**Comment by CarlosCRG19** [comment]
we might want to limit the floating point to 3 decimal places (?)

File: server/adapters/object_detection.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            boxes = result.boxes
+
+            for box in boxes:
+                class_code = int(box.cls[0])
+                class_name = self.model.names[class_code]
+
+                confidence = float(box.conf[0])
+
+                x1, y1, x2, y2 = box.xyxy[0]
+                x1, y1, x2, y2 = float(x1), float(y1), float(x2), float(y2)
```
Comment (BrandonMagana) [inline server/adapters/object_detection.py:49]: NIT, maybe we could round these floating points to 2-3 decimal places. This is merely a personal preference and the fact that we don't need that extra precision, haha.

**Review by BrandonMagana — APPROVED** [review]
Looks all rigth to me

**Comment by BrandonMagana** [comment]
Also, are we going to keep the `weights.pt` file on the repo?

File: server/adapters/object_detection.py (hunk 1 above)
Comment (CarlosCRG19) [reply server/adapters/object_detection.py:49]: You are right on this, lemme make the changes

**Comment by CarlosCRG19** [comment]
> Also, are we going to keep the `weights.pt` file on the repo?

@BrandonMagana currently working on a version that does not require the weights file stored in the repo. I think its better to store the weights on S3 and fetch them when building the image

**Pull request merged**